src/packet.rs

```diff
@@ -1,17 +1,40 @@
 //! MC Protocol packets
 
-use std::old_io::{ IoError, IoErrorKind, IoResult };
+use byteorder::{BigEndian, ReadBytesExt, WriteBytesExt};
+
+use std::error::FromError;
+use std::io;
+use std::io::prelude::*;
 
 use types::VarInt;
 
 use uuid::Uuid;
 
+pub trait ReadExactExt: Read {
```

Comment:
This should go in a separate file, shouldn't it? It's also used for String. Maybe `ioutils.rs` and reexported on `lib.rs`.